Change to single quotes

tjamesmac opened the pull request:
### What is the context of this PR?
### Talk to Andy Pitt before deploying to live so he can test the changes.

Quotes are killing google analytics in live

Live surveys don't have an `ga` working right now because the cookies `HTML` isn't getting pulled in.

We are checking for the presence of double-quotes to pull in `ga` where RAS are using single quotes

This means that in live surveys, if a user accepts cookies in RAS, `ga` isn't getting pulled in. Where as, if cookies are accepted within a survey, `ga` does get pulled in.

We need a solution to take into account both cases.

Sort of a big deal because rolling back breaks GDPR compliance and not doing anything means there is no analytics on any surveys.

#### Tom B edit 26/11, changes:

- Change our JS to emit single quotes in case downstream are expecting single to come back (since they initially set single-quote cookies)
- Refactor the cookie JS and python slightly so the quote-setting logic is just in one function in case we want to change it again
- Change runner so it's happy with either double or single quotes - it now converts to & parses the JSON for the cookie

tjamesmac committed:
Change to single quotes

zimboflyman reviewed (approved):
looks l;ike you've covered all the usage: true spots!

tbaggaley committed:
Merge branch 'master' into EAR-1043-cookie-quote

tbaggaley committed:
🛠 Allow GA cookies to have single or double quotes

Refactors out analytics allowed checking code in runner into its own
function so can be modified in one place if need be. This function
parses the cookie as JSON, first replacing single with double quotes in
order to allow this to happen.

It now works with both double and single quotes.

tbaggaley committed:
🧹Fix linting

tbaggaley committed:
🧹 More linting...

Use single quotes instead of double for strings
Remove else branch

tbaggaley committed:
🛠 Make our JS emit single quoted cookies

Also refactors some of the JS code to use utility functions so the logic
is in one place - only have to change once then if we later decide to
all use double quotes

tbaggaley committed:
✅ Add test for analytics_allowed function

tbaggaley committed:
✅ Actually run the new test (d'oh)

tbaggaley committed:
Merge branch 'master' into EAR-1043-cookie-quote
